application/models/Flights.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * Dummy model class that represent our airline's flight schedule
+ */
+class Flights extends CI_Model
+{
+    // The mock flight data
+    var $data = array(
+        // L_caravan routes
+        'L_YPRZMT' => array(
+            'departsFrom' => 'L_YPR', //airport id
+            'arrivesAt' => 'L_ZMT',   //airport id
+            'departureTime' => '0800',
+            'arrivalTime' => '0900',
+            'plane' => 'L_caravan'),  //plane id
+        'L_ZMTYZP' => array(          //... and so on
+            'departsFrom' => 'L_ZMT',
+            'arrivesAt' => 'L_YZP',
+            'departureTime' => '1300',
+            'arrivalTime' => '1400',
+            'plane' => 'L_caravan'),
+        'L_YZPYPR' => array(
+            'departsFrom' => 'L_YZP',
+            'arrivesAt' => 'L_YPR',
+            'departureTime' => '1800',
+            'arrivalTime' => '1900',
+            'plane' => 'L_caravan'),
+
+        // L_kingair routes
+        'L_YPRYZP' => array(
+            'departsFrom' => 'L_YPR',
+            'arrivesAt' => 'L_YZP',
+            'departureTime' => '0900',
+            'arrivalTime' => '1000',
+            'plane' => 'L_kingair'),
+        'L_YZPZMT' => array(
+            'departsFrom' => 'L_YZP',
+            'arrivesAt' => 'L_ZMT',
+            'departureTime' => '1400',
+            'arrivalTime' => '1500',
+            'plane' => 'L_kingair'),
+        'L_ZMTYPR' => array(
+            'departsFrom' => 'L_ZMT',
+            'arrivesAt' => 'L_YPR',
+            'departureTime' => '1900',
+            'arrivalTime' => '2000',
+            'plane' => 'L_kingair'),
+
+        // pc12ng routes
+        'L_YPRYXT1' => array(
+            'departsFrom' => 'L_YPR',
+            'arrivesAt' => 'L_YZP',
+            'departureTime' => '0800',
+            'arrivalTime' => '0900',
+            'plane' => 'L_pc12ng'),
+        'L_YXTYPR1' => array(
+            'departsFrom' => 'L_YZP',
+            'arrivesAt' => 'L_ZMT',
+            'departureTime' => '1300',
+            'arrivalTime' => '1400',
+            'plane' => 'L_pc12ng'),
+        'L_YPRYXT2' => array(
+            'departsFrom' => 'L_ZMT',
+            'arrivesAt' => 'L_YPR',
+            'departureTime' => '1800',
+            'arrivalTime' => '1900',
+            'plane' => 'L_pc12ng'),
+        'L_YXTYPR2' => array(
```

Comment:
This flight doesn't seem possible because the previous flight already flew from ZMT to YPR